src/components/login/forms/register-form.jsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+import ArrowButton from './../../arrow-button/arrow-button';
+import Input from './../../common/input';
+import EmailIcon from './../../icons/email-icon';
+import PadlockIcon from './../../icons/padlock-icon';
+
+import validations from './../../../validations';
+
+import { postJson } from './../../../utils/ajax';
+
+const FormFooter = styled.footer`
+    align-items: center;
+    display: flex;
+    justify-content: space-between;
+`;
+
+const Link = styled.button`
+    background: none;
+    border: none;
+    color: #ccc;
+    cursor: pointer;
+    font: 1.6em/1.5 Oxygen;
+
+    &:hover {
+        text-decoration: underline;
+    }
+`;
+
+const ErrorMessage = styled.div`
+    font-family: Oxygen;
+    font-size: 1rem;
+    color: red;
+`;
+
+export default class RegisterForm extends React.Component {
+    static propTypes = {
+        showLoginForm: React.PropTypes.func,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
```

Comment:
I think redux-form perfectly suites handling forms like this - it contains validation